src/rollup/index.ts

```diff
@@ -223,8 +227,21 @@ export default function rollup (rawInputOptions: GenericConfigObject) {
 
 		timeStart('--BUILD--');
 
-		return graph.buildSingle(inputOptions.input)
-			.then((bundle) => {
+		const codeSplitting = inputOptions.experimentalCodeSplitting;
+
+		if (codeSplitting) {
+			if (typeof inputOptions.input === 'string')
+				inputOptions.input = [inputOptions.input];
+		}
+		else if (inputOptions.input instanceof Array && !codeSplitting) {
+			error({
+				code: 'INVALID_OPTION',
+				message: 'Multiple inputs only supported when setting the experimentalCodeSplitting flag option.'
+			});
+		}
+
+		if (!codeSplitting) return graph.buildSingle(inputOptions.input)
```

Comment:
When the codeSplitting flag is true then the TS types returned by generate will be different.  Currently TS believes that the same types are returned whether this is set or not.  I am not exactly sure the right approach to resolve this rather than setting an OR statement in the OutputBundle's generate result.

Reply:
There are basically two overloads of the generic rollup function from a typing perspective:

```js
interface SingleInputOptions extends InputOptions {
  input: string;
}
interface MultipleInputOptions extends InputOptions {
  input: string[];
}
function rollup(options: SingleInputOptions): Promise<BundleSingle>;
function rollup(options: MultipleInputOptions): Promise<BundleChunks>;
```

Reply:
Cool this would work as well.